### PR TITLE
Add Hydra dev docs

### PR DESCRIPTION
**Pull request by impactmass**
Resolves reactioncommerce/reaction-hydra#6
Type: **docs**

## Changes:
- Adds docs on how Ory Hydra is used as part of Reaction platform services
- Explains how a client configuration is added for the storefront
- Explains how the Identity Provider features work
- Adds links to Hydra Production guides

### CODE REVIEW

**Comment by machikoyasuda** [comment]
Where do you think this doc should live on the Table of Contents?

I am thinking of we should:
- Make a `Hydra` section and add this there as the first doc on Hydra.
- Add a link to this page and write a short blurb in the Storefront README
- Add a link to this page on the Platform README

What do you think @impactmass @ticean?

**Comment by impactmass** [comment]
I think we can add a part of it under: `Developer docs in "Getting Started as a Developer" section`. The deployment part can go under the "Deploying section". Then a link in the hydra readme to those public docs. Eric had suggested this on the [ticket](https://github.com/reactioncommerce/reaction-hydra/issues/6#issuecomment-424878940). 

I'll split them up when I get notes on deployment from Grig.

File: public-docs/understanding-hydra-setup.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+Hydra does not offer user management (it does not provide user sign up, user log in, password reset flow), but instead uses a redirection-based flow and a REST API to connect with an existing identity provider.
+
+We’ve setup Hydra to use our existing Meteor application as the Identity provider. This gives us the benefits of OAuth2 and OpenID Connect while allowing us to use our existing user system.
+
+Read more about Hydra, OAuth2 and OpenID Connect on Hydra docs [here](https://www.ory.sh/docs/guides/master/hydra/).
+
+
+### OAuth 2.0 Client Setup
+
+We’ve setup the reaction-hydra repository with a sample client creation script. This script registers an OAuth 2.0 Client for the sample storefront ([reaction-next-starterkit](https://github.com/reactioncommerce/reaction-next-starterkit)) into Hydra.
```
[the remark this answers is not in the document]
Comment (impactmass) [reply public-docs/understanding-hydra-setup.md]: 👍 Started the discussion https://github.com/reactioncommerce/reaction-next-starterkit/issues/310#issuecomment-427083029

**Comment by impactmass** [comment]
@griggheo reminder to add the deploy comments

**Pull request merged**